crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -1635,6 +1635,11 @@ where
 
         emit!(n.key);
 
+        // emit for a computed property, but not an identifier already marked as optional
+        if n.is_optional && !n.key.as_ident().map(|i| i.optional).unwrap_or(false) {
+            punct!("?");
+        }
```

Comment:
When parsed it will have `optional` as `true` for both `n.is_optional` and on the ident, so that's why this condition is here. Somewhat related: https://github.com/swc-project/swc/issues/7279